### ISSUE TITLE
Add a new permission to expose chrome.ipfs API to extensions

### ISSUE THREAD

**Issue by bbondy** [brave/brave-browser#13998]
Add a new permission for `chrome.ipfs` api and notify the user of this permission when they try to install a Chrome extension that has this permission.

Example for an extension that requests the permission "ipfs" we should list that they are requesting it here:
<img width="889" alt="Screen Shot 2021-03-17 at 4 08 34 PM" src="https://user-images.githubusercontent.com/831718/111531517-07495a00-873b-11eb-8ba3-3c9980f84ecf.png">

This extension ID is currently published to the chrome store with the `ipfs` permission for testing:
`fpeoodllldobpkbkabpblcfaogecpndd`

https://chrome.google.com/webstore/detail/webrecorder-archivewebpag/fpeoodllldobpkbkabpblcfaogecpndd

This is a talk by the same author that would like this API:
https://www.youtube.com/watch?v=evcSETnTBf0

so basically the task is in 2 parts, and 1) is needed before we are allowed to do 2)

1. Show IPFS info in that screenshot above about IPFS
2. Give chrome.ipfs access to any extension that requests it

**Comment by stephendonner** [comment]
Verified `PASSED` using the testplan from https://github.com/brave/brave-core/pull/8289

Brave | 1.24.47 Chromium: 90.0.4430.41 (Official Build) nightly (x86_64)
-- | --
Revision | e9c92b1eaca8487e212f3f6bc081fdb6d4863759-refs/branch-heads/4430@{#723}
OS | macOS Version 11.2.3 (Build 20D91)

Steps:

1.  Installed extension with ipfs permission in the manifest file, for example `https://chrome.google.com/webstore/detail/webrecorder-archivewebpag/fpeoodllldobpkbkabpblcfaogecpndd`
2.  Confirmed alert about IPFS in permission bubble
<img width="1312" alt="Screen Shot 2021-04-05 at 11 52 26 AM" src="https://user-images.githubusercontent.com/387249/113613350-f5eec180-9605-11eb-8b9b-2b303ab516ae.png">

3.  Went to `chrome://inspect#extensions` and inspected Webrecorder background page
<img width="1312" alt="Screen Shot 2021-04-05 at 12 00 10 PM" src="https://user-images.githubusercontent.com/387249/113613754-8200e900-9606-11eb-9f36-bd6fe22a0f5c.png">

4.  Checked that namespace chrome.ipfs has only 3 public methods available
<img width="752" alt="Screen Shot 2021-04-05 at 11 40 32 AM" src="https://user-images.githubusercontent.com/387249/113613891-ae1c6a00-9606-11eb-866f-7a907fc95086.png">

5.  Checked the same for Brave Shield extension, all ipfs methods should be available
<img width="885" alt="Screen Shot 2021-04-05 at 11 48 23 AM" src="https://user-images.githubusercontent.com/387249/113613894-aeb50080-9606-11eb-9a73-4194160dcf58.png">
